examples/ui/directional_navigation.rs

```diff
@@ -0,0 +1,417 @@
+
+// This observer will be triggered whenever a button is pressed
+// In a real project, each button would also have its own unique behavior,
+// to capture the actual intent of the user
+fn universal_button_click_behavior(
+    // We're using a on-mouse-down trigger to improve responsiveness;
+    // Clicked is better when you want roll-off cancellation
+    mut trigger: Trigger<Pointer<Pressed>>,
+    mut button_query: Query<(&mut BackgroundColor, &mut ResetTimer)>,
+) {
+    let button_entity = trigger.target();
+    if let Ok((mut color, mut reset_timer)) = button_query.get_mut(button_entity) {
+        // This would be a great place to play a little sound effect too!
+        color.0 = PRESSED_BUTTON.into();
+        reset_timer.0 = Timer::from_seconds(0.3, TimerMode::Once);
+
+        // Picking events propagate up the hierarchy,
+        // so we need to stop the propagation here now that we've handled it
+        trigger.propagate(false);
+    }
+}
+
+/// Resets a UI element to its default state when the timer has elapsed.
+#[derive(Component, Default, Deref, DerefMut)]
+struct ResetTimer(Timer);
+
+fn reset_button_after_interaction(
+    time: Res<Time>,
+    mut query: Query<(&mut ResetTimer, &mut BackgroundColor)>,
+) {
+    for (mut reset_timer, mut color) in query.iter_mut() {
+        reset_timer.tick(time.delta());
+        if reset_timer.just_finished() {
+            color.0 = NORMAL_BUTTON.into();
+        }
+    }
+}
+
+// We're spawning a simple 3x3 grid of buttons
+// The buttons are just colored rectangles with text displaying the button's name
+fn setup_ui(
+    mut commands: Commands,
+    mut directional_nav_map: ResMut<DirectionalNavigationMap>,
+    mut input_focus: ResMut<InputFocus>,
+) {
+    const N_ROWS: u16 = 5;
+    const N_COLS: u16 = 3;
+
+    // Rendering UI elements requires a camera
+    commands.spawn(Camera2d);
+
+    // Create a full-screen background node
+    let root_node = commands
+        .spawn(Node {
+            width: Val::Percent(100.0),
+            height: Val::Percent(100.0),
+            ..default()
+        })
+        .id();
+
+    // Add instruction to the left of the grid
+    let instructions = commands
+        .spawn((
+            Text::new("Use arrow keys or D-pad to navigate. \
+            Click the buttons, or press Enter / the South gamepad button to interact with the focused button."),
+            Node {
+                width: Val::Px(200.0),
+                justify_content: JustifyContent::Center,
+                align_items: AlignItems::Center,
+                margin: UiRect::all(Val::Px(16.0)),
+                ..default()
+            },
+        ))
+        .id();
+
+    // Set up the root entity to hold the grid
+    let grid_root_entity = commands
+        .spawn(Node {
+            display: Display::Grid,
+            // Allow the grid to take up the full height and width of the window
+            width: Val::Percent(60.),
```

Comment:
IMO

- Make instructions node wider, maybe `Px(300.0)` 
- Leave this at `100.0`.
- Adjust this comment to say "take up the full height and the rest of the width."

Seems okay for the grid to not be centered in the screen to me.